src/components/Sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import { NavLink } from "react-router-dom";
+import { Link, LinkGroup } from "./types";
+import Icon from "components/Icon";
+import { createNavLinkStyler } from "helpers";
+import Header from "./Header";
+
+type Props = {
+  endowId: number;
+  className?: string;
+  linkGroups: LinkGroup[];
+  onChange?: (link: Link) => void;
+};
+
+export default function Sidebar(props: Props) {
```

Comment:
> @ap-justin as is already said in a couple of previous comments (https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/1716#discussion_r1102353311 and https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/1716#discussion_r1104108641), the example usage is in the PR description.

> The Sidebar is a "dumb" component that just displays the links passed into it. If any links need to be conditionally displayed, updated, hidden etc. then this functionality would be the responsibility of the component containing the Sidebar.

As I have also repeated a couple of times, i know this but let me elaborate

> it's dumb

yes no question about this

> just displays the links passed into it

If you have actually tried to use this in the existing pages you would see a big difference on the data that needs to be passed. in your `isolated` example you are passing `whole`  `RouteData`

```ts

 type Link = {
  title: string;
  to: string;
  icon: {
    type: IconType;
    size: number;
  };
};

```
but since I tried to use this component in `/admin` I realized that using `<Sidebar/>` on different admin groups this way would mean duplicating some of `type Link` since admin types have common routes `proposals`, `admin-wallet`, `settings`

so I opt to embedd `type Link` to `adminRoutes` and just pass to `paths` to `<Sidebar/>`

```tsx
//core
<Layout linkGroups={[
  {
    links: ["proposals"],
  },
]}/>`

//review
<Layout linkGroups={ links: ["index_review", "proposals"] } />

//charity
<Layout [
  {
    links: ["index", "withdraws"],
  },
  {
    title: "Invest",
    links: ["invest", "settings"],
  },
  { title: "Profile", links: ["edit_profile"] },
  { title: "Manage", links: ["proposals"] },
]; /> 

```

on top of this, I realized there's even no need to have `<Sidebar/>` on `/components` since its current use case is just in `Admin/Layout`.

Reply:
@ap-justin 
> If you have actually tried to use this in the existing pages you would see a big difference on the data that needs to be passed. in your isolated example you are passing whole type Link

The example just uses some dummy data, not sure what "whole type Link" means?

> but since I tried to use this component in /admin I realized that using <Sidebar/> on different admin groups this way would mean duplicating some of type Link since admin types have common routes proposals, admin-wallet, settings

Comparing the implementation of `Sidebar` here and your draft PR, the implementation is the same, I don't see how the original (here) is suboptimal?

(note: the only difference I saw was that you removed the `Link.icon` field even though different links need different icons)

> so I opt to embedd type Link to adminRoutes and just pass to paths to <Sidebar/>

Yes, you could do that with this Sidebar version too.

> on top of this, I realized there's even no need to have <Sidebar/> on /components since its current use case is just in Admin/Layout

Sure, but that does not explain why the `Sidebar` here is suboptimal in any way.

Reply:
@misicnenad 

> Where exactly did you mention you actually saw the example?

If I haven't saw you are requesting reviewers to make `placeholder` usage - I woudn't even request you to just use this component with the `actual` use case.

> The example just uses some dummy data, you could pass whatever you want to it anyway.
Not sure what "whole type Link" means?

you created this component and you don't know your `types` ? so I could just pass `string[]` to `type Link[]` - is that what you are saying? 

 > (note: the only difference I saw was that you removed the Link.icon field even though different links need different icons)
 
 this is not even the point, if you would just observe: is that I don't pass metadata to `<Sidebar/>` i just pass string paths to it and let it get the metadata
 
 > Yes, you could do that with this Sidebar version too.
 
please look closely the difference it's just `type Link` vs `string` (path name)


 > It still does not explain why the Sidebar here is suboptimal in any way.
 
 If I haven't used this component I wouldn't see that you could just pass `/pathname` to it and 
 * without `:id` or full route, determine `match` with `react-router-dom/matchPath`
 * Sidebar could just pullout metadata from a map so even multiple pages could just pass the same path without duplicating metadata
 
 ```
 //core
<Layout linkGroups={[
  {
    links: ["proposals"], 
  },
]}/>`

//review
<Layout linkGroups={ links: ["index_review", "proposals"] } />
 ```
 * that `Sidebar` being treated as part of `admin/Layout` - implementation could be of lower level `createElement` so that there's no need for `MobileSidebar`


 
 

Reply:
> > Where exactly did you mention you actually saw the example?
>
> If I haven't saw you are requesting reviewers to make placeholder usage - I woudn't even request you to just use this component with the actual use case.

Again you do not answer my question. You have a habit of ignoring my questions. 
But ok, it doesn't even matter.

>  If I haven't saw you are requesting reviewers to make placeholder usage - I woudn't even request you to just use this component with the actual use case.

What? You found problems with me adding an example usage of the sidebar??

> you created this component and you don't know your [types](https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/1716/files#diff-14728898de288a6435afc4e7974651993523e96da242917307af46f8a796aa97R3) ? so I could just pass string[] to type Link[] - is that what you are saying?

No. What I am saying is that you use unclear language when writing comments and what you meant with "whole type Link" is unclear.

> this is not even the point, if you would just observe: is that I don't pass metadata to <Sidebar/> i just pass string paths to it and let it get the metadata

Sure, that's one way to implement this. And it brings some tradeoffs with it too - now `Sidebar` is tightly connected with `admin/*` pages and cannot be reused on other pages.
If you're 100% sure it will never be used outside of `admin/*` pages (@SovereignAndrey will it?), then sure, this can be one way to implement route passing (to sidebar). But nothing is 100% in software dev. 

You could've done the same with the previous `Sidebar` version and place the metadata extraction into the parent component.

> please look closely the difference it's just type Link vs string (path name)

I see now. See comment(s) above for feedback.

> * without :id or full route, determine match with react-router-dom/matchPath

Could've done that with this version of `Sidebar` too.

> Sidebar could just pullout metadata from a map so even multiple pages could just pass the same path without duplicating metadata

You make `Sidebar` tightly connected to `admin/*` pages this way, making it unusable by other pages in the future (if necessary). But if we're sure it won't be used elsewhere @SovereignAndrey , then ok.

> that Sidebar being treated as part of `admin/Layout` - implementation could be of lower level `createElement` so that there's no need for `MobileSidebar`

This is just an alternative version of the same thing implemented here, with a one less file/component and one more element-creation complexity. I'm fine with either one.

Reply:
@misicnenad 

> What? You found problems with me adding an example usage of the sidebar?? 

That's what's this thing is all about - I have requested you so many times to just use this component in the actual pages rather than giving `example`. But you always say no "placeholders are good enough"

> Sure, that's one way to implement this. And it brings some tradeoffs with it too - now Sidebar is tightly connected with admin/* pages and cannot be reused on other pages.
If you're 100% sure it will never be used outside of admin/* pages

See, by `actually` using this component brings light into useful discussions like this. Even if this `<Layout/>` is going to be used into another management page like instead of `/aif/{id}` instead `/admin/{id}` most of routes would still hold usable 
```
//all
  "index",
  "proposal",
  "proposals",
  "templates",
  "settings",
  "admin-wallet",

  //npo | aif
  "edit_profile",

  //npo
  "widget_config",
  "withdraws",
  "invest",

  //ap
  "index_review",
```

> This is just an alternative version of the same thing implemented here, with a one less file/component and one more element-creation complexity. I'm fine with either one.

this is confusing, how could you say its the same while mentioning differences - which I would iterate is now visible because this component is `actually` used and not just rendered in `placeholder/isolated` manner

Reply:
Ok, now that actual implementation discussion is taking place, let's stop running in circles with the I-said-this-you-said-that discussions, it's unproductive. 

As I asked in one of previous comments https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/1716#discussion_r1107272688, can we transfer https://github.com/AngelProtocolFinance/angelprotocol-web-app/issues/1720 to Justin since he practically implemented it already @SovereignAndrey ? 
We can also co-assign him to https://github.com/AngelProtocolFinance/angelprotocol-web-app/issues/1762 as well, since he appropriated the `Sidebar` from this PR into that draft one. That way we close this PR and both the sidebar and the admin updates will be merged together.

Reply:
@misicnenad, demo PR is just to show Sidebar usage on existing admin pages - it's incomplete (without proper `icon` or `iconSize`)

Reply:
> @misicnenad I think it would be good if you could build off the work done in this branch for your followup page shuffling task. Doing that would give us a fuller view of how this new sidebar will work with all the various admins menus and would clean up the placeholder AIF stuff. Since there's no Figma design for the AP Team or the Applications admins, we can simply replicate the existing Top Navs for each in the side bar, posing minimal changes for those areas.

Will start work on this.